README.md

```diff
@@ -35,54 +35,107 @@ pod 'Argo', :git => 'https://github.com/thoughtbot/Argo'
 
-### Git Submodules ###
+### Git Submodules
 
 I guess you could do it this way if that's your thing.
 
 Add this repo as a submodule, and add the project file to your workspace. You
 can then link against `Argo.framework` for your application target.
 
-## Usage
-
-First, create your model. We like to use structs but a class is OK too.
+## Usage tl;dr:
 
 ```swift
 struct User {
   let id: Int
   let name: String
   let email: String?
+  let role: Role
+  let friends: [User]
 }
-```
 
-Then, extend the model to conform to `JSONDecodable`. You will also need to
-write a static constructor method that is curried. We like to use `create`.
-
-```swift
 extension User: JSONDecodable {
-  static func create(id: Int)(name: String)(email: String?) -> User {
-    return User(id: id, name: name, email: email)
+  static func create(id: Int)(name: String)(email: String?)(role: Role)(friends: [User]) -> User {
+    return User(id: id, name: name, email: email, role: role, friends: friends)
+  }
+
+  static func decode(j: JSONValue) -> User? {
+    return User.create
+      <^> j <| "id"
+      <*> j <| "name"
+      <*> j <|? "email" // Use ? for parsing optional values
+      <*> j <| ["role", "title"] // parse nested attributes
+      <*> j <|| "friends" // parse arrays of objects
+  }
+}
+
+// Wherever you receive JSON data:
+
+let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions(0), error: .None)
+
+if let j: AnyObject = json {
+  if let value = JSONValue.parse(j) {
+    let user = User.decode(value)
   }
 }
 ```
 
-It's important that the `create` function's parameters are all in separate
-parenthesis. This will make the function curried. Then inside the function we
-can call the model's constructor.
+## Ideology
 
-Finally, implement the `JSONDecodable` `decode` function to decode the incoming
-JSON. Here is an example of decoding a `User` from JSON received from a network
-request.
+Argo's core concept is that in order to maintain type safety, you should only
+be able to successfully decode an object if all parameters are satisfied
+properly. So if you have a model that looks like this:
 
-JSON received from network request:
+```swift
+struct User {
+  let id: Int
+  let name: String
+}
 ```
+
+but the JSON you receive from the server looks like this:
+
+```json
 {
-  "id": 1,
-  "name": "Cool User",
-  "email": "cool.user@example.com"
+  "user": {
+    "id": "this isn't a number",
+    "name": "Gob Bluth"
+  }
 }
 ```
 
-Parse data to JSON then to a `JSONValue` and finally to our `User` object:
+you would want that JSON parsing to fail and return `.None` instead of a
+`User` object. That base concept means that when you're dealing with an actual
+`User` object, you can be sure that when you ask for `user.id`, you're going
+to get the type you expect.
+
+## Functional Concepts
+
+Argo really wants to be used with patterns stolen from Functional Programming
```

Comment:
nit pick: stolen feels to negative .. how about "seen in" instead of "stolen from" or "used in"